datalad/support/gitrepo.py

```diff
@@ -996,20 +996,13 @@ def config(self):
             self._cfg = ConfigManager(dataset=self, dataset_only=False)
         return self._cfg
 
-    def is_with_annex(self, only_remote=False):
-        """Return True if GitRepo (assumed) at the path has remotes with git-annex branch
-
-        Parameters
-        ----------
-        only_remote: bool, optional
-            Check only remote (no local branches) for having git-annex branch
-        """
-        return any((b.endswith('/git-annex') or
-                    'annex/direct' in b
-                    for b in self.get_remote_branches())) or \
-            ((not only_remote) and
-             any((b == 'git-annex' or 'annex/direct' in b
-                  for b in self.get_branches())))
+    def is_with_annex(self):
+        """Report if GitRepo (assumed) has (remotes with) a git-annex branch
+        """
+        return any(
+            b['refname:strip=2'] == 'git-annex' or b['refname:strip=2'].endswith('/git-annex')
+            for b in self.for_each_ref_(fields='refname:strip=2', pattern=['refs/heads', 'refs/remotes'])
```

Comment:
Because we don't care about the actual output, this could be further simplified to

```diff
diff --git a/datalad/support/gitrepo.py b/datalad/support/gitrepo.py
index 2cee19aee..71a6d84ae 100644
--- a/datalad/support/gitrepo.py
+++ b/datalad/support/gitrepo.py
@@ -1000,8 +1000,7 @@ def is_with_annex(self):
         """Report if GitRepo (assumed) has (remotes with) a git-annex branch
         """
         return any(
-            b['refname:strip=2'] == 'git-annex' or b['refname:strip=2'].endswith('/git-annex')
-            for b in self.for_each_ref_(fields='refname:strip=2', pattern=['refs/heads', 'refs/remotes'])
+            self.for_each_ref_(fields='refname', pattern=['refs/*/git-annex'])
         )
 
     @classmethod
```

We're offloading more of the processing to git, so that shouldn't hurt performance, but, at least with our benchmarks, I didn't see any improvements with this change.